### ISSUE TITLE
Warning for last + after

### ISSUE THREAD

**Issue by amccloud** [facebook/relay#247]
I'm getting this warning:

```
GraphQLRange does not currently handle retrieval for before(<cursor>).first(<count>) and after(<cursor>).last(<count>)
```

For this query fragment:

```
fragment on User {
  photos(last: $count) {
    edges {
      node {
        id,
        imageUri,
        createdAt
      }
    }
  }
}
```

Which appears to convert to:

```
query DiscoverController{node(id:"VXNlcjox"){..._0u97mo2}} fragment _0u97mo2 on User{_photosyhsn3z:photos(before:"YXJyYXljb25uZWN0aW9uOjI0",after:"YXJyYXljb25uZWN0aW9uOjI=",last:8){edges{node{id,imageUri,createdAt},cursor},pageInfo{hasNextPage,hasPreviousPage}},id}
```

**Comment by amccloud** [comment]
I should note that `first: $count` works fine.

**Comment by josephsavona** [comment]
cc @yuzhi

**Comment by yuzhi** [comment]
Thanks for reporting and providing such complete information.
TLDR: You can probably ignore the warning.

Looking at the query it is trying to send to the server, it has both a before and after call. That can only happen when we have a gap of unknown edges between two segment of known edges. 
```
edgeA
edgeB
edgeC
.
.
.
(unfetched edges )
.
.
.
edgeX
edgeY
edgeZ
```

My best guess is you have some components fetching the `first:n` of this photos connection and some components fetching the `last:m` of this same photos connection, which generates the gap and that query. That totally within what Relay's connection manager should support.

The warning is mainly to prevent people from directly doing `before/first` and `after/last` inside components. Those two combinations are used for polling additional things off of a connection and should not be part of any component's queries. The warning is showing up because we recently updated the write path to also read out some edges for some additional logic.

@josephsavona  do you think we can move that warning about doing `before/first` and `after/last` to the transform step so we don't get a lot of false positive warnings?

**Comment by josephsavona** [comment]
> do you think we can move that warning about doing before/first and after/last to the transform step so we don't get a lot of false positive warnings?

Awesome idea, I'm on it.

**Comment by amccloud** [comment]
> My best guess is you have some components fetching the first:n of this photos connection and some components fetching the last:m of this same photos connection, which generates the gap and that query. That totally within what Relay's connection manager should support.

That is correct :)

**Comment by yuzhi** [comment]
@josephsavona I can remove the warning from GraphQLRange once you update the transform.

**Comment by josephsavona** [comment]
@yuzhi sounds perfect

**Comment by yuzhi** [comment]
Removed it from GraphQLRange https://github.com/facebook/relay/commit/a04a51486c2576c4a0dcdb091023e1bda0d0a5fe
